storm-core/test/jvm/org/apache/storm/scheduler/resource/TestResourceAwareScheduler.java

```diff
@@ -54,6 +61,681 @@
+        builder2.setSpout("wordSpoutX", new TestWordSpout(), 1);
+        builder2.setSpout("wordSpoutY", new TestWordSpout(), 1);
+        StormTopology stormTopology2 = builder2.createTopology();
+        Map<ExecutorDetails, String> executorMap2 = TestUtilsForResourceAwareScheduler.genExecsAndComps(stormTopology2, 1, 0);
+        TopologyDetails topology2 = new TopologyDetails("topology2", config, stormTopology2, 0, executorMap2, 0);
+
+        Cluster cluster = new Cluster(iNimbus, supMap, new HashMap<String, SchedulerAssignmentImpl>(), config);
+        ResourceAwareScheduler rs = new ResourceAwareScheduler();
+
+        Map<String, TopologyDetails> topoMap = new HashMap<>();
+        topoMap.put(topology1.getId(), topology1);
+        topoMap.put(topology2.getId(), topology2);
+        Topologies topologies = new Topologies(topoMap);
+
+        rs.prepare(config);
+        rs.schedule(topologies, cluster);
+
+        SchedulerAssignment assignment1 = cluster.getAssignmentById(topology1.getId());
+        Set<WorkerSlot> assignedSlots1 = assignment1.getSlots();
+        Set<String> nodesIDs1 = new HashSet<>();
+        for (WorkerSlot slot : assignedSlots1) {
+            nodesIDs1.add(slot.getNodeId());
+        }
+        Collection<ExecutorDetails> executors1 = assignment1.getExecutors();
+
+        Assert.assertEquals(1, assignedSlots1.size());
+        Assert.assertEquals(1, nodesIDs1.size());
+        Assert.assertEquals(7, executors1.size());
+        Assert.assertEquals("Running - Fully Scheduled by DefaultResourceAwareStrategy", cluster.getStatusMap().get(topology1.getId()));
+
+        SchedulerAssignment assignment2 = cluster.getAssignmentById(topology2.getId());
+        Set<WorkerSlot> assignedSlots2 = assignment2.getSlots();
+        Set<String> nodesIDs2 = new HashSet<>();
+        for (WorkerSlot slot : assignedSlots2) {
+            nodesIDs2.add(slot.getNodeId());
+        }
+        Collection<ExecutorDetails> executors2 = assignment2.getExecutors();
+
+        Assert.assertEquals(1, assignedSlots2.size());
+        Assert.assertEquals(1, nodesIDs2.size());
+        Assert.assertEquals(2, executors2.size());
+        Assert.assertEquals("Running - Fully Scheduled by DefaultResourceAwareStrategy", cluster.getStatusMap().get(topology2.getId()));
+    }
+
+    @Test
+    public void testTopologySetCpuAndMemLoad() {
+        INimbus iNimbus = new TestUtilsForResourceAwareScheduler.INimbusTest();
+        Map<String, Number> resourceMap = new HashMap<>();
+        resourceMap.put(Config.SUPERVISOR_CPU_CAPACITY, 400.0);
+        resourceMap.put(Config.SUPERVISOR_MEMORY_CAPACITY_MB, 2000.0);
+        Map<String, SupervisorDetails> supMap = TestUtilsForResourceAwareScheduler.genSupervisors(2, 2, resourceMap);
+
+        TopologyBuilder builder1 = new TopologyBuilder(); // a topology with multiple spouts
+        builder1.setSpout("wordSpout", new TestWordSpout(), 1).setCPULoad(20.0).setMemoryLoad(200.0);
```

Comment:
Hi @jerrypeng , the original clj test only tests slot and executor number where the cpu/mem load number does not actually matter. This test set a specific number for cpu and mem so that we can verify the assigned memory and cpu in the end (either 110 or 200 will work).
+        Assert.assertEquals(400.0, assignedMemory, 0.001);
+        Assert.assertEquals(40.0, assignedCpu, 0.001);